File: src/game/components/upgrades.lua
local constants = require 'game.components.constants'

local Upgrades = {
  _meta = constants.Upgrades
}

function Upgrades.new(id, passive, throw, dash, grapple, dig)
  return {
    _meta = Upgrades._meta,
    id = id,
    passive = passive or 0,
    throw = throw or 0,
    dash = dash or 0,
    grapple = grapple or 0,
    dig = dig or 0
  }
end

return Upgrades

File: client2/src/game/components/upgrades.ts
import { ComponentFlag } from '../flags'

export class Upgrades {
  static _id = 'Upgrades'
  _id = Upgrades._id

  static _flag = ComponentFlag.Upgrades
  _flag = ComponentFlag.Upgrades

  passive: number
  throw: number
  dash: number
  grapple: number
  dig: number

  constructor(
    passive: number,
    throw_: number,
    dash: number,
    grapple: number,
    dig: number
  ) {
    this.passive = passive
    this.throw = throw_
    this.dash = dash
    this.grapple = grapple
    this.dig = dig
  }
}
